### PR TITLE
allows user paths for ca_cert verify path

### PR DESCRIPTION
**Pull request by a-tal**
currently you get the following error:

```SSLError("bad ca_certs: '~/example.crt'", Error([('system library', 'fopen', 'No such file or directory'), ('BIO routines', 'BIO_new_file', 'no such file'), ('x509 certificate routines', 'X509_load_cert_crl_file', 'system lib')],))```

### CODE REVIEW

**Comment by Lukasa** [comment]
Thanks for this!

Can we apply this change to `ca_cert_dir` as well?

File: urllib3/connection.py [hunk 1]
```diff
@@ -205,11 +206,17 @@ def set_cert(self, key_file=None, cert_file=None,
...
         self.cert_reqs = cert_reqs
-        self.ca_certs = ca_certs
-        self.ca_cert_dir = ca_cert_dir
         self.assert_hostname = assert_hostname
         self.assert_fingerprint = assert_fingerprint
 
+        if ca_certs:
+            ca_certs = os.path.expanduser(ca_certs)
+        if ca_cert_dir:
+            ca_cert_dir = os.path.expanduser(ca_cert_dir)
```
Comment (shazow) [inline urllib3/connection.py]: Hm this line needs test coverage now. :) Could rewrite it as one line if we'd prefer.

```diff
- self.ca_certs = ca_certs
- self.ca_cert_dir = ca_cert_dir
+self.ca_certs = ca_certs and os.path.expanduser(ca_certs)
+self.ca_cert_dir = ca_cert_dir and os.path.expanduser(ca_cert_dir)
```

Comment (a-tal) [inline urllib3/connection.py]: I'm good with that :+1:

**Comment by shazow** [comment]
Thank you! :custard:

**Comment by shazow** [comment]
Now that I think about it, we really don't expand any path settings anywhere else, not sure how to feel about that. :x

**Pull request merged**